bot/moderation/moderation.py

```diff
@@ -22,17 +26,540 @@ def __init__(self, bot):
+
+    @tempmute_user.error
+    @tempban_user.error
+    async def temp_action_error(self, ctx: commands.Context, error: commands.CommandError):
+        """Error Handler for commands imposing temporary punishments.
+
+        Handles an exception which occurs if the specified amount of time is invalid.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            error (commands.CommandError): The error raised during the execution of the command.
+        """
+        if isinstance(error, commands.CommandInvokeError) and isinstance(error.original, ValueError):
+            await ctx.send("**__Error:__** Die angegebene Zeitdauer ist ungültig. :clock330:")
+
+    @commands.command(name='kick', hidden=True)
+    @command_log
+    async def kick_user(self, ctx: commands.Context, user: discord.Member, *, reason: Optional[str]):
+        """Command Handler for the `kick` command.
+
+        Kicks the specified member from the server.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            user (discord.Member): The member who should be kicked.
+            reason (Optional[str]): The reason provided by the moderator.
+        """
+        await user.kick(reason=reason)
+        await ctx.send(f"{user.mention} wurde gekickt. :anger:")
+
+        embed = _build_mod_action_embed("Kick", f"Du wurdest von **__{self.guild}__** gekickt.", ctx.author, reason)
+        await user.send(embed=embed)
+
+    @commands.command(name='namehistory', hidden=True, aliases=["aka"])
+    @command_log
+    async def member_nicknames(self, ctx: commands.Context, *, user: discord.Member):
+        """Command Handler for the `namehistory` command.
+
+        Posts an embed listing all the different names which the specified member has used on the server. Additional
+        timestamps representing when the members name has been changed are also provided.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            user (discord.Member): The member whose names are being requested.
+        """
+        nicknames = self._db_connector.get_member_names(user.id)
+        description = "Es werden maximal die __letzten {0} Namen__ eines Mitglieds angezeigt, welche auf diesem " \
+                      "Server verwendet wurden." \
+            .format(constants.LIMIT_NICKNAMES)
+
+        if nicknames:
+            embed = discord.Embed(title=f"Namensverlauf von {user} :page_with_curl:", description=description,
+                                  color=constants.EMBED_COLOR_MODERATION, timestamp=datetime.utcnow())
+            embed.set_footer(text="Stand")
+            embed.set_thumbnail(url=user.avatar_url)
+            embed.add_field(name=user.display_name, value="aktuell")
+
+            for name in nicknames[:constants.LIMIT_NICKNAMES]:
```

Comment:
Why is the limit static?

What if a user had a bad nickname and you want to show that to someone, but he changed it a few times since you realized and you can no longer view the bad nickname without entering the db by hand. 
If it was configurable you could use it for edge cases to show (and prove) bad nicknames.

Reply:
Discord embeds as well as normal messages have a character limit which is why I have to limit the amount of entries shown. A solution for this would be some kind of pagination which is kinda tedious to realize for Discord messages. The developer of discord.py is already working on some kind of implementation for his library so my suggestion is to simply wait until its being released by him. I also don't believe that we will need this kind of functionality anytime soon and if we do it would be pretty obvious if someone tries to trick as you described anyway.